File: basic-calculator-ii/basic-calculator-ii.cs
public class Solution {
    public int Calculate(string s) {
        var stack = new Stack<int>();
         char op = '+';
        int num =0;
        
        for(int i=0;i<=s.Length;i++)
        {
            char c = i<s.Length ? s[i] : '+';
            if(c == ' ') continue;
            
            if(char.IsNumber(c))
            {
                num = num *10 + (c-'0');
            }
            else
            {
                if(op =='+') stack.Push(num);
                if(op =='-') stack.Push(-num);
                if(op=='*') stack.Push(stack.Pop() * num);
                if(op =='/') stack.Push(stack.Pop() / num);
                
                op = c;
                num =0;
            }
        }
        
        return stack.Sum();
        
    }
}

File: Basic Calculator II .java
class Solution {
    public int calculate(String s)
    {

        int res = 0, num = 0;

        char opr = '+';

        Stack<Integer> stack = new Stack<Integer>();

        for(char chr : (s + "+").toCharArray()) {

            if(chr == ' ') continue;

            if(Character.isDigit(chr)) num = num * 10 + (chr - '0');

            else {

            if(opr == '+') stack.push(num);

            else if(opr == '-') stack.push(-num);

            else if(opr == '*') stack.push(stack.pop() * num);

            else if(opr == '/') stack.push(stack.pop() / num);

            num = 0;

            opr = chr;

            }

        }

        while(!stack.isEmpty()) res += stack.pop();

        return res;
}}
